viewing_party/party.py

```diff
@@ -0,0 +1,227 @@
+# ************   Wave 1   ************
+def create_movie(title, genre, rating):
+    # return a dictionary if all parameters are Truthy.
+    # return None if any parameter is Falsy
+
+    if title == None or genre == None or rating == None:
+        return None
+
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    """
+    i.e user_data = {"watched": [{"title": "Title A", "genre": "Horror", "rating": 3.5}]}
+    """
+    # an empty list that user has no movies to watch
+    # add movie to "watched" list inside of user_data
+    user_data["watched"].append(movie)
+
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+def watch_movie(user_data, title):
+    # user_data = {"watchlist": [{title: " "...}]
+    #             "watched": [{title: " "...}]}
+
+    for movie in user_data["watchlist"]:
+        # {title: " "...}
+        if movie["title"] == title:
+            #" "
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+
+    return user_data
```

Comment:
Be careful about modifying a collection as you're iterating through it (watchlist). For this example, we're only removing a single instance of a movie, however, if there are multiple instances of that movie that could cause the rest of the items in the list to shift position and potentially skip over other items as the loop continues.

This case is fine, but let's proceed with caution when modifying a collection while iterating through it.

Here's more info: https://thispointer.com/python-remove-elements-from-a-list-while-iterating/